core/tests/protractor_utils/editor.js

```diff
@@ -372,28 +387,85 @@ var addParameterChange = function(paramName, paramValue) {
 
 // RULES
 
-// This function selects a rule for the current interaction and enters the
-// entries of the parameterValues array as its parameters; the parameterValues
-// should be specified after the ruleName as additional arguments. For example
-// with interaction 'NumericInput' and rule 'Equals' then there is a single
-// parameter which the given answer is required to equal.
-var _selectRule = function(ruleElement, interactionId, ruleName) {
+var selectRuleInAddResponseModal = function(interactionId, ruleName) {
+  var ruleElement = element(by.css('.protractor-test-add-response-details'));
+  _selectRule(ruleElement, interactionId, ruleName);
+};
+
+var selectRuleParametersInAddResponseModal = function() {
+  var ruleElement = element(by.css('.protractor-test-add-response-details'));
+  var args = [ruleElement];
+  for (var i = 0; i < arguments.length; i++) {
+    args.push(arguments[i]);
+  }
+  _selectRuleParameters.apply(null, args);
+};
+
+// From the corresponding ruleDescription string deduces and returns the types
```

Comment:
Parses the relevant ruleDescription string, and returns an Array containing the types of the rule input parameters.